lib/kafo/configuration.rb

```diff
@@ -49,11 +49,11 @@ def self.get_scenario_id(filename)
       File.basename(filename, '.yaml')
     end
 
-    def initialize(file, persist = true)
+    def initialize(file, logger, persist = true)
```

Comment:
How about keeping the API compatible:
```suggestion
    def initialize(file, persist = true, logger = KafoConfigure.logger)
```

I'm also not sure if we can rely on this (I think it may be Ruby 2.5+), but you can also enforce a named argument:
```suggestion
    def initialize(file, persist = true, logger:)
```

Or with a default:
```suggestion
    def initialize(file, persist = true, logger: KafoConfigure.logger)
```

Then you must call it as:
```ruby
Configuration.new(config, logger: KafoConfigure.logger)
```